src/test/ui/self/arbitrary_self_types_nested.rs

```diff
@@ -0,0 +1,36 @@
+// run-pass
```

Comment:
A compile-fail test should check non-object-safety?

Reply:
There already is one-- src/test/ui/self/arbitrary-self-types-not-object-safe.rs/.stderr